src/state/StateNode.java

```diff
@@ -142,8 +143,6 @@ public boolean equals(Object other) {
             if (vals == null) {
                 break;
             }
-            Collections.sort(vals);
```

Comment:
I made a mistake and we still need this. The sort is on the values (which are lists) rather than the keys.